### ISSUE TITLE
Add the tracked value which can show the dimension collapse problem  

### ISSUE THREAD

**Issue by AlekseySh** [OML-Team/open-metric-learning#32]
We may follow the idea from LeCun's paper and use the percentage of "big" singular values in SVD
Technically we may put it inside the `EmbeddingMetric`

Another approach is to track how many principal components in PCA we need to keep 95% of accuracy. (What is the policy for the case when PCA improves accuracy?)

**Comment by dapladoc** [comment]
Hi! I would like to tackle this problem. As I see it, we need to add to `oml/functional/metrics.py` a function called like

```python
def calc_topological_metrics(embeddings: torch.Tensor, **kwargs) -> TMetrics
```

and inside it we can compute different non-retrieval metrics, like PCA-based, EMD-based, etc.

Following this issue, in the `calc_topological_metrics` there will be just one function call at the moment with the following signature

```python
def calc_linear_dimension(embeddings: torch.Tensor, n_components: Tuple[int, ...]) -> List[torch.Tensor]
```

The `calc_linear_dimension` takes `embeddings`, and `n_components` --- a list the desired amount of information to preserve during linear projection (in percents). As an output it gives a list of the numbers of components that is sufficient to keep `n_components` information (absolute value or in percents too?).

See [sklearn.decomposition.PCA](https://scikit-learn.org/stable/modules/generated/sklearn.decomposition.PCA.html) for the reference.

We can actually make `calc_linear_dimension` return some additional information, like all the singular values, and log it as it is in form of a histogram, or just a plots.

What do you think it?

**Comment by dapladoc** [comment]
Just in case, here is the link on the LeCun's paper [Understanding Dimensional Collapse in Contrastive Self-supervised Learning](https://arxiv.org/abs/2110.09348)

**Comment by dapladoc** [comment]
By the way, maybe it worth to add an issue to implement the _DirectCLR_ proposed in this paper as a head of any other backbone?

**Comment by dapladoc** [comment]
> I like the proposed design. Just want to double-check: is it right that you want to call `calc_topological_metrics` somewhere nearby `calc_retrieval_metrics` in `EmbeddingsMetrics` class? If so, it's okay.

Yes, `calc_topolotical_metrics` will be called near the `calc_retrieval_metrics` in `EmbeddingMetrics`.



> The only thing that I would change is the current confusing signature. What about this?
> `def count_meaningful_components(embeddings, examplained_variance_to_keep: Tuple[int, ...]) -> n_components`
> Or you can suggest other ideas.

I'm ok with this naming.
